### PR TITLE
Move to Native Brotli, TypeScript, Require Node >= 10.16.0

### PR DESCRIPTION
**Pull request by kristoferbaxter**
This PR shouldn't be merged until the project is comfortable with requiring a version of Node supporting native Brotli.

Work is happening to try and move Brotli support into the LTS Node 10 version. (cc @MylesBorins)

### CODE REVIEW

**Comment by siddharthkp** [comment]
Brotli support was added to Node 10.16.0: https://nodejs.org/docs/latest-v10.x/api/zlib.html

Would this mean a breaking change to this repo? (deprecating support for < 10.16.0)

**Comment by kristoferbaxter** [comment]
@MylesBorins – Is there a recommended pattern for falling back from installed via core to a dependency (and not requiring the dependency upfront during installation)?

**Comment by MylesBorins** [comment]
@kristoferbaxter AFAIK there is not such pattern. npm has "optional dependencies". What I would suggest is to drop support for Node 8 in a semver major and keep that older version around with some basic support. If folks are using < 10.16.0 then can install that version

**Comment by kristoferbaxter** [comment]
@erwinmombay – I've updated this PR based on @siddharthkp and @MylesBorins feedback.

It now requires Node > `10.16.0`, and bumped the semver to `2.0.0`.

I'd recommend bumping the current master to `1.0.0` and releasing it so `1.0.0` and `2.0.0` can be used by authors of differing node environment versions.

**Comment by ForsakenHarmony** [comment]
would love to see this land so we can include it in [microbundle](https://github.com/developit/microbundle)

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    [brotliConstants.BROTLI_PARAM_MODE]: passed && 'mode' in passed && passed.mode || brotliConstants.BROTLI_DEFAULT_MODE,
+    [brotliConstants.BROTLI_PARAM_QUALITY]: passed && 'quality' in passed && passed.quality || brotliConstants.BROTLI_MAX_QUALITY,
+    [brotliConstants.BROTLI_PARAM_SIZE_HINT]: toEncode ? toEncode.byteLength : 0,
+  }
+});
+
+/**
+ * @param incoming Either a Buffer or string of the value to encode.
+ * @param options Subset of Encoding Parameters.
+ * @return
```
Comment (erwinmombay) [inline src/index.ts]: `@return Promise`

**Review by erwinmombay — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (kristoferbaxter) [reply src/index.ts]: Resolved in 054c97b.

**Comment by ForsakenHarmony** [comment]
🎉

**Pull request merged**